rust/src/dataset/scanner.rs

```diff
@@ -93,12 +96,42 @@ impl Scanner {
             offset: None,
             nearest: None,
             with_row_id: false,
+            fragment: None,
```

Comment:
what's the purpose of a scanner that only does 1 fragment?

Reply:
This is to support `pyarrow.dataset.Fragment.to_batch()`. 
So via exposing `DataFragment` using pyarrow API, users can get a list of Fragments on different machine, and each Fragment can be used as basic unit to read a file, distributedly.